runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -106,7 +108,46 @@ void J9::RecognizedCallTransformer::transform(TR::TreeTop* treetop)
       case TR::java_lang_Math_min_L:
          processIntrinsicFunction(treetop, node, TR::lmin);
          break;
+      case TR::java_lang_StringUTF16_toBytes:
+         process_java_lang_StringUTF16_toBytes(treetop, node);
+         break;
       default:
          break;
       }
    }
+
+void J9::RecognizedCallTransformer::process_java_lang_StringUTF16_toBytes(TR::TreeTop* treetop, TR::Node* node)
+   {
+   TR_J9VMBase* fej9 = static_cast<TR_J9VMBase*>(comp()->fe());
+
+   TR::Node* valueNode = node->getChild(0);
+   TR::Node* offNode = node->getChild(1);
+   TR::Node* lenNode = node->getChild(2);
+
+   anchorAllChildren(node, treetop);
+   prepareToReplaceNode(node);
+
+   int32_t byteArrayType = fej9->getNewArrayTypeFromClass(reinterpret_cast<TR_OpaqueClassBlock*>(fej9->getJ9JITConfig()->javaVM->byteArrayClass));
+
+   TR::Node::recreateWithoutProperties(node, TR::newarray, 2, 
+      TR::Node::create(TR::ishl, 2,
```

Comment:
Are there any edge cases to worry about here when lenNode represents a value that could overflow if shifted ?

Reply:
I initially had the same question, but I found it might be the limitation by Java itself. The length node of `TR::newarray` is 32-bit, which complies Java specs.
Moreover, `java/lang/StringUTF16.toBytes()` takes a char array and returns a byte array. The maximum possible size of a char array is MAX_INT*sizeof(char) = 2*MAX_INT, the maximum possible size of a byte array is only MAX_INT. The method itself cannot handle the case where `TR::ishl` overflows.

Reply:
Actually, after more thinking about this, we may need to check the original behavior of `java/lang/StringUTF16.toBytes()` when the input array has length greater than MAX_INT/2. The transformed trees should behave the same as the original one.

Reply:
So the Java code itself will throw a `NegativeArraySizeException` when the overflow occurs and it is the exact same exception we'll throw from the VM helper. So I don't think there is an issue here with overflow.

Reply:
Okay as long as the behaviour is the same whether this JIT transformation is done or not (which is what you confirmed), that answers my question.